storage.go

```diff
@@ -4,37 +4,39 @@ import (
 	"context"
 )
 
-// Storage represents the abstraction for underlying storage backends. A storage
-// implementation does not care about specific data types. All the
-// storage cares about are key-value pairs. Code making use of the storage
-// have to take care about specific types they care about them self.
+// KV is a key-value pair.
+type KV struct {
+	// Key should take a format of path separated by slashes "/". E.g.
+	// "/a/b/c". Storage implementations should validate the key before
+	// storing them. Key sanitized before inserting to the storage. I.e.
+	// leading slash can be added and trailing slash can be removed. E.g.
+	// "/a/b/c", "a/b/c/", "a/b/c/", and "/a/b/c/" represent the same key.
+	Key string
+	// Val is an arbitrary value stored under the Key.
+	Val string
+}
+
+func NewKV(key, val string) KV {
```

Comment:
Yes, something like this would work I'd be very happy about it. 

```
type KV struct {
	key, val string
}

func NewKV(key, val string) (KV, error) {
	err := validate(key)
	if err == nil {
		return KV{}, err
	}

	return KV{key,val}, nil
}

func (k KV) Key() string { return k.key }
func (k KV) Val() string { return k.val }
```

There is a little nuance though.

Say you have data like this in a storage:

```
/foo/bar
/foo/baz
```

ATM if you `List("/")` you'd receive `["foo/bar", "foo/baz"]`. But if we started using KV with validation built in you'd receive `["/foo/bar", "/foo/baz"]`. Or we have to do some sneaky migration in current `kubernetesd` TPOs.